torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -158,20 +174,110 @@ struct MemoryFormat {
     // storing stride_order in `permuted_order` for a simpler life, so we don't
     // have to decode `permutation_` when we want to apply/restore permutation_.
     permuted_order_ = stride_order;
-    bool has_permutation_ = false;
+    bool has_permutation = false;
     for (const auto i : c10::irange(rank)) {
       permutation_ = permutation_ * 10 + stride_order[i];
```

Comment:
what if someone calls `setPermutation` twice? With `permutation_` as a class member this will lead to weird results

Reply:
:face_with_head_bandage: I'll reset permutation_ to 0.